src/ext/java/org/opentripplanner/ext/legacygraphqlapi/LegacyGraphQLUtils.java

```diff
@@ -29,12 +29,19 @@ public static Locale getLocale(DataFetchingEnvironment environment, String local
       return Locale.forLanguageTag(localeString);
     }
 
+    // This can come from the accept-language header
+    var envLocale = environment.getLocale();
+
     Map<String, ?> localContext = environment.getLocalContext();
-    if (localContext != null && localContext.get("locale") != null) {
+    if (
+      (envLocale == null || envLocale.getLanguage().equals("*")) &&
+      localContext != null &&
+      localContext.get("locale") != null
+    ) {
       return (Locale) localContext.get("locale");
     }
 
-    return environment.getLocale();
+    return envLocale;
```

Comment:
This is is a bit awkward, what happens if both `envLocale` and `localContext` is null?

I would write this:

```
public class GraphQLUtils {

  public static Locale getLocale(DataFetchingEnvironment environment) {
    // This can come from the accept-language header
    var userLocale = environment.getLocale();
    var defaultLocale = localContextLocale(environment);

    if(userLocale == null) {
      return defaultLocale.orElse(Locale.forLanguageTag("?????"));
    }

    if (defaultLocale.isPresent() && acceptAny(userLocale)) {
      return defaultLocale.get();
    }
    return userLocale;
  }

  private static Optional<Locale> localContextLocale(DataFetchingEnvironment environment) {
    Map<String, ?> localContext = environment.getLocalContext();
    if(localContext == null) {
      return Optional.empty();
    }
    return Optional.ofNullable((Locale) localContext.get("locale"));
  }

  private static boolean acceptAny(@Nonnull Locale locale) {
    return locale.getLanguage().equals("*");
  }
}

```